lib/mapObjects/MiscObjects.cpp

```diff
@@ -195,6 +196,15 @@ ui32 CGMine::defaultResProduction() const
 	}
 }
 
+ui32 CGMine::getProducedQuantity() const
+{
+	auto * playerSettings = cb->getPlayerSettings(getOwner());
+	auto ret = producedQuantity * playerSettings->handicap.percentIncome / 100;
+	if(ret == 0 && producedQuantity > 0) // create at least 1 resource
+		ret = 1;
+	return ret;
```

Comment:
Maybe always round up instead?
```suggestion
	// always round up income - we don't want mines to always produce zero if handicap in use
	return (producedQuantity * playerSettings->handicap.percentIncome + 99) / 100;
```